api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -0,0 +1,246 @@
+ *
+ * Created by wpride1 on 8/11/15.
+ */
+public class SqlHelper {
+    public static void dropTable(Connection c, String storageKey) {
+        String sqlStatement = "DROP TABLE IF EXISTS " + storageKey;
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            Logger.log("E", "Could not drop table: " + e.getMessage());
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    public static void createTable(Connection c, String storageKey, Persistable p) {
+        String sqlStatement = DatabaseHelper.getTableCreateString(storageKey, p);
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            //fail silently, expected sometimes
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    public static PreparedStatement prepareIdSelectStatement(Connection c,
+                                                             String storageKey,
+                                                             int id) {
+        try {
+            PreparedStatement preparedStatement =
+                    c.prepareStatement("SELECT * FROM " + storageKey + " WHERE "
+                            + DatabaseHelper.ID_COL + " = ?;");
+            preparedStatement.setInt(1, id);
+            return preparedStatement;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static PreparedStatement prepareTableSelectStatement(Connection c,
+                                                                String storageKey,
+                                                                String[] fields,
+                                                                String[] values,
+                                                                Persistable p) {
+        org.commcare.modern.database.TableBuilder mTableBuilder =
+                new org.commcare.modern.database.TableBuilder(storageKey);
+        mTableBuilder.addData(p);
+        Pair<String, String[]> mPair = DatabaseHelper.createWhere(fields, values, p);
+        try {
+            String queryString =
+                    "SELECT * FROM " + storageKey + " WHERE " + mPair.first + ";";
```

Comment:
Ah, I see, these aren't user inputs, they are code inputs. Probably OK in this case, but please make a note in the argslist for the method